components/dashboard/src/components/ide-settings.tsx

```diff
@@ -38,47 +66,32 @@ export class IDESettings extends React.Component<IDESettingsProps> {
     }
 
     private renderRadio(label: string, value: IDEKind) {
-        const checked = value === this.value;
+        const checked = value === this.state.value;
         return <Grid item xs={12}>
-            <FormControlLabel control={<Radio color="default" />} label={label} value={value} checked={checked} onChange={this.updateDefaultIde} />
-            {value === 'image' && <Input value={this.image} onChange={this.updateDefaultIde} />}
+            <FormControlLabel control={<Radio color="default" />} label={label} value={value} checked={checked} onChange={this.updateState} />
+            {value === 'image' && <Input value={this.state.image} onChange={this.updateState} />}
         </Grid>;
     }
 
-    private get value(): IDEKind {
-        const defaultIde = this.props.user.additionalData?.ideSettings?.defaultIde;
-        if (isIDEAlias(defaultIde)) {
-            return defaultIde;
-        }
-        if (defaultIde) {
-            return 'image';
-        }
-        return 'theia';
-    }
-
-    private get image(): string | undefined {
-        const defaultIde = this.props.user.additionalData?.ideSettings?.defaultIde;
-        if (isIDEAlias(defaultIde)) {
-            return undefined;
-        }
-        return defaultIde;
-    }
-
-    private updateDefaultIde = (event: React.ChangeEvent<HTMLInputElement>) => {
-        let value = this.value;
-        let image = this.image;
+    private updateState = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const state: IDESettingsState = {}
         if (event.target.type === 'radio') {
-            value = event.target.value as IDEKind;
+            state.value = event.target.value as IDEKind;
         } else {
-            image = event.target.value;
+            state.image = event.target.value;
         }
+        this.setState(state, () => this.fireStateChange());
+    }
+
+    private fireStateChange(): void {
+        const { value, image } = this.state;
 
         const additionalData = (this.props.user.additionalData || {});
         const settings = additionalData.ideSettings || {};
         if (value === 'theia') {
             delete settings.defaultIde;
         } else if (value === 'image') {
-            settings.defaultIde = image;
+            settings.defaultIde = image || '';
```

Comment:
if someone tried to start a workspace with this setting, i.e. just selected image and didn't enter an actual image, workspace starts fail without an error message.

We should add validation here to prevent this case.

Reply:
or fall back to theia alias instead?

Reply:
it didn't for me. I was able to just select image and leave it at that.
When I tried to start a workspace the startup broke - as expected.